pkg/api/norman/customization/aks/handler.go

```diff
@@ -243,6 +244,10 @@ func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credI
 	if region != "" {
 		cap.ResourceLocation = region
 	}
+	currentVersion := req.URL.Query().Get("currentVersion")
```

Comment:
I know we're following the exact same way `region` is assigned, but really what's the point of `if` check?
```go
cap.CurrentVersion = req.URL.Query().Get("currentVersion")
```
This should be perfectly fine. Even if we consider a hypothetical case where a pointer to `Capabilities` struct with some set fields is passed, I would imagine we would want `CurrentVersion` to be set to empty if `req` doesn't have  `currentVersion` query parameter passed.

Oh, and I just noticed the check on line below checks `region` instead of `currentVersion`. :)

Reply:
I agree there is no need to do this assignment I was just following the pattern above.

Reply:
👍  Feel free to follow the pattern if you see some value in it, but if you do we need to have the `if` check fixed.

Reply:
Oops, I had thought we had gotten rid of these unnecessary empty checks in earlier revisions of this code. I don't think there's value in following the pattern, direct assignment should be fine.